yb-voyager/src/datastore/s3Datastore.go

```diff
@@ -55,20 +55,18 @@ func (ds *S3Datastore) FileSize(filePath string) (int64, error) {
 }
 
 // filepath.Join converts URL (s3://...) to directory-like string (s3:/...).
-func (ds *S3Datastore) Join(elem ...string) string {
+func (ds *S3DataStore) Join(elem ...string) string {
 	finalPath := ""
-	for _, fragment := range elem {
-		finalPath += fragment + "/"
-	}
-	return finalPath[:len(finalPath)-1]
+	finalPath += strings.Join(elem, "/")
+	return finalPath
 }
 
-func (ds *S3Datastore) Open(resourceName string) (io.ReadCloser, error) {
+func (ds *S3DataStore) Open(resourceName string) (io.ReadCloser, error) {
 	// resourceName is hidden underneath a symlink for s3 objects.
 	s3Resource, err := os.Readlink(resourceName)
 	if err != nil {
 		utils.ErrExit("unable to resolve symlink %v to s3 resource: %v", resourceName, err)
 	}
-	reader, err := s3.DownloadS3Object(s3Resource)
+	reader, err := s3.OpenReadableObject(s3Resource)
```

Comment:
Change `s3Resource` to `objectPath`.

Change `OpenReadableObject()` to something like `NewObjectReader(objectPath)` or `OpenObjectForReading(objectPath)`.